libs/elasticsearch-core/src/main/java/org/elasticsearch/bootstrap/JarHell.java

```diff
@@ -68,25 +67,23 @@ private JarHell() {}
     @SuppressForbidden(reason = "command line tool")
     public static void main(String args[]) throws Exception {
         System.out.println("checking for jar hell...");
-        checkJarHell();
+        checkJarHell(System.out::println);
         System.out.println("no jar hell found");
     }
 
     /**
      * Checks the current classpath for duplicate classes
+     * @param output A {@code String} {@link Consumer} to which debug output will be sent
      * @throws IllegalStateException if jar hell was found
      */
-    public static void checkJarHell() throws IOException, URISyntaxException {
+    public static void checkJarHell(Consumer<String> output) throws IOException, URISyntaxException {
         ClassLoader loader = JarHell.class.getClassLoader();
-        Logger logger = Loggers.getLogger(JarHell.class);
-        if (logger.isDebugEnabled()) {
-            logger.debug("java.class.path: {}", System.getProperty("java.class.path"));
-            logger.debug("sun.boot.class.path: {}", System.getProperty("sun.boot.class.path"));
-            if (loader instanceof URLClassLoader ) {
-                logger.debug("classloader urls: {}", Arrays.toString(((URLClassLoader)loader).getURLs()));
-             }
+        output.accept("java.class.path: " + System.getProperty("java.class.path"));
+        output.accept("sun.boot.class.path: " + System.getProperty("sun.boot.class.path"));
+        if (loader instanceof URLClassLoader ) {
```

Comment:
Nit, and I know it was there, but there's an extra space between `URLClassLoader` and `)`.